### PR TITLE
Remove loop over groups in the custom regularizer (for semi-sparse group lasso)

### PR DESCRIPTION
**Pull request by xrenard**
In SSGL_WeightRegularizer a loop over the groups was used to compute
the loss function (regularization term on groups).

Since the number of features can be very large (>1000) in the context
of the Random Shapelet -and thus the number of groups- the training was
very long. The loop is replaced by tensorflow.segment_sum function and
the prepare_groups function is now useless.

The training time is now in the seconds instead of 10+ minutes.

In discussion with Alexis Gatignol.

### CODE REVIEW

**Comment by rtavenar** [comment]
Hi,

Great to hear from you about this!
Reading your code made me realize that the original implementation does not correspond to the Simon et al. paper in which L2 penalty is applied instead of L2^2 in our implementation (which removes the group effect).

Could you fix this as well in your merge? This would imply using K.sqrt(...) before summing in your l.267. Also, I don't know if it's easy to get the sqrt(p_l) as in the paper (Eq. 3) :/ This is not a big issue for the moment as for our case, all p_l are equal, but would be nice if we could fix it.
